Source/Shaders/PolylineShadowVolumeMorphVS.glsl

```diff
@@ -0,0 +1,158 @@
+attribute vec3 position3DHigh;
+attribute vec3 position3DLow;
+
+attribute vec4 startHi_and_forwardOffsetX;
+attribute vec4 startLo_and_forwardOffsetY;
+attribute vec4 startNormal_and_forwardOffsetZ;
```

Comment:
Is there anything that relates the start, end, and right normals so that we could use fewer vertex attributes, and then recompute them here with high precision?

Reply:
I don't think so, the start and end normals help define planes containing the geodesic surface normals at the start and end points of the line segment on the ellipsoid. They also point towards the next and previous line segments.

@bagnell and I discussed this, it would take either a lot more vertex attributes or geometry shader support if we wanted to compute all the vectors on the shader side.